src/lib/mac-address.ts

```diff
@@ -0,0 +1,49 @@
+import * as _ from 'lodash';
+import * as fs from 'mz/fs';
+import * as path from 'path';
+
+import { shouldReportInterface } from '../network';
+import log from './supervisor-console';
+
+export class MacAddressError extends Error {}
+
+export async function getAll(sysClassNet: string): Promise<string | undefined> {
+	try {
+		// read the directories in the sysfs...
+		const interfaces = await fs.readdir(sysClassNet);
+
+		return _(
+			await Promise.all(
+				interfaces.map(async (intf) => {
+					try {
+						const [addressFile, typeFile, masterFile] = [
+							'address',
+							'type',
+							'master',
+						].map((f) => path.join(sysClassNet, intf, f));
+						const intfType = (await fs.readFile(typeFile, 'utf8')).trim();
+						const intfHasMaster = await fs.exists(masterFile);
+						if (
+							!shouldReportInterface(intf) ||
+							intfType !== '1' ||
+							intfHasMaster
+						) {
+							return '';
+						}
+
+						const addr = await fs.readFile(addressFile, 'utf8');
+						return addr.split('\n')[0].trim().toUpperCase();
+					} catch {
+						return '';
+					}
+				}),
+			),
+		)
+			.filter((addr) => addr !== '')
+			.sortedUniq()
```

Comment:
This requires the array to be sorted or otherwise it won't work - I don't see anything that guarantees the array will be sorted:
```ts
_.sortedUniq([1, 1, 2]); // [1, 2]
_.sortedUniq([1, 2, 1]); // [1, 2, 1]